### PR TITLE
Send: Amount field [Missing Error state]

### PR DESCRIPTION
**Pull request by KalervoHyyppa**
## Description

Displays error field in amount section in Send native coins

DEPENDS ON RIBN-TOOLKIT PR [RIBN-735](https://github.com/Topl/ribn-toolkit/pull/67)

Fixes # (issue)

[RIBN-735](https://topl.atlassian.net/jira/software/c/projects/RIBN/boards/5?modal=detail&selectedIssue=RIBN-735)

## Type of change

Please delete options that are not relevant.

-   [x] New feature (non-breaking change which adds functionality)

## Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings

[RIBN-735]: https://topl.atlassian.net/browse/RIBN-735?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/presentation/transfers/poly_transfer_section.dart [hunk 1]
```diff
@@ -121,42 +100,47 @@ class _PolyTransferSectionState extends State<PolyTransferSection> {
...
-                    handleResult: (bool result) {
-                      setState(() {
+                  onChanged: (text) {
+                    ref.read(polyTransferProvider.notifier).state =
+                        ref.read(polyTransferProvider).copyWith(recipientAddress: text);
+                    validateRecipientAddress(
+                      networkName: vm.currentNetwork.networkName,
+                      address: _recipientController.text,
+                      handleResult: (bool result) {
                         if (result) {
```
Comment (ultimaterex) [inline lib/presentation/transfers/poly_transfer_section.dart:122]:
```
if (result) {
  polyNotifier.state = polyProvider.copyWith(validRecipientAddress: _recipientController.text);
  _recipientController.text = '';
} else {
  polyNotifier.state =polyProvider.copyWith(validRecipientAddress: '');
}
```
is much more readable imo

File: lib/presentation/transfers/poly_transfer_section.dart [hunk 2]
```diff
@@ -121,42 +100,47 @@ class _PolyTransferSectionState extends State<PolyTransferSection> {
...
                   controller: _recipientController,
                   validRecipientAddress: _validRecipientAddress,
                   // validate the address entered on text change
-                  onChanged: (text) => validateRecipientAddress(
-                    networkName: widget.vm.currentNetwork.networkName,
-                    address: _recipientController.text,
-                    handleResult: (bool result) {
-                      setState(() {
+                  onChanged: (text) {
+                    ref.read(polyTransferProvider.notifier).state =
```
[the remark this answers is not in the document]
Comment (ultimaterex) [reply lib/presentation/transfers/poly_transfer_section.dart:116]: Also just replace the ref.reads with the variable accessor you define up in build

**Review by ultimaterex — CHANGES_REQUESTED** [review]

File: ios/fastlane/report.xml [hunk 3]
```diff
@@ -5,12 +5,12 @@
     
     
       
-      <testcase classname="fastlane.lanes" name="0: default_platform" time="0.000165">
+      <testcase classname="fastlane.lanes" name="0: default_platform" time="0.000228">
```
[the remark this answers is not in the document]
Comment (KalervoHyyppa) [reply ios/fastlane/report.xml:8]: Good call will do

**Review by ultimaterex — APPROVED** [review]
Looks good!

**Pull request merged**